runtime/buffer.h

```diff
@@ -16,6 +16,9 @@ namespace slinky {
 #ifdef __APPLE__
 using index_t = std::int64_t;
 static_assert(sizeof(index_t) == sizeof(std::size_t));
+#elif defined __EMSCRIPTEN__
+using index_t = std::int64_t;
+// Note that size_t and ptrdiff_t are sizeof(int32_t) under Emscripten
```

Comment:
Shouldn't index_t be std::int32_t then?

Reply:
Slinky won't compile properly unless index_t is a 64-bit type, or so it would appear based on my experimenting so far. This is a bit of a band-aid that I want to revisit after getting some other pieces working.